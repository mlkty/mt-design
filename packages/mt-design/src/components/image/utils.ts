
export const isValidImage = (src: string) => {
    return new Promise(resolve => {
        const image = new Image();
        image.onload = () => resolve(true);
        image.onerror = () => resolve(false);
        image.src = src;
    });
};
