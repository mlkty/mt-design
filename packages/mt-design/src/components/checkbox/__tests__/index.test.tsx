import {Checkbox} from '..';
import {customRender, userEvent} from '../../../../tests/utils';
import mountTest from '../../../../tests/mountTest';

describe('Checkbox', () => {
    mountTest(Checkbox);

    it('render checkbox correctly', async () => {
        const {container} = customRender(<Checkbox />);

        const element = container.firstChild as HTMLElement;
        expect(element).toHaveClass('mt-checkbox', {exact: true});
        await userEvent.click(element);
        expect(element).toHaveClass('mt-checkbox mt-checkbox--checked', {exact: true});
    });

    it('render disabled checkbox', async () => {
        const {container} = customRender(<Checkbox disabled />);
        const element = container.firstChild as HTMLElement;

        expect(element).toHaveClass('mt-checkbox mt-checkbox--disabled', {exact: true});
        await userEvent.click(element);
        expect(element).toHaveClass('mt-checkbox mt-checkbox--disabled', {exact: true});
    });

    it('render disabled checked checkbox', async () => {
        const {container} = customRender(<Checkbox defaultChecked disabled />);
        const element = container.firstChild as HTMLElement;

        expect(element).toHaveClass('mt-checkbox mt-checkbox--checked mt-checkbox--disabled', {exact: true});
        await userEvent.click(element);
        expect(element).toHaveClass('mt-checkbox mt-checkbox--checked mt-checkbox--disabled', {exact: true});
    });

    it('render onChange checkbox', async () => {
        const onChange = jest.fn();

        const {container} = customRender(<Checkbox onChange={onChange} />);
        const element = container.firstChild as HTMLElement;

        await userEvent.click(element);
        expect(onChange).toBeCalledWith(true, expect.anything());
    });
});
