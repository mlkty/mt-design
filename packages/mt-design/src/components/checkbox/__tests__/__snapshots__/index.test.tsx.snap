// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render checkbox classname correctly 1`] = `
<div>
  <label
    class="mt-checkbox"
    role="checkbox"
  >
    <input
      type="checkbox"
    />
    <span
      class="mt-checkbox-input"
    />
    <span
      class="mt-checkbox-inner"
    />
  </label>
</div>
`;
